### PR TITLE
Tape mode

### PR DESCRIPTION
**Pull request by antalszava**
Prepares the plugin to be compatible with tape mode:

* [X] Updates the gradient tests to use the `requires_grad` argument
* [X] Deprecates the QPU and raises an error
* [X] Disables the converter and raises an error

This PR branches off of #67.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 pyquil>=2.16
 git+https://github.com/PennyLaneAI/pennylane.git
 networkx
-flaky
+flaky
```
Comment (antalszava) [inline requirements.txt:4]: newline change

File: pennylane_forest/converter.py [hunk 2]
```diff
@@ -327,7 +327,9 @@ class ProgramLoader:
     _matrix_dictionary = pyquil.simulation.matrices.QUANTUM_GATES
 
     def __init__(self, program):
-        self.program = program
+        raise ValueError("The PyQuil program conversion feature is currently not supported.")
```
Comment (josh146) [inline pennylane_forest/converter.py:330]: :+1:

It might be worth removing the device card on the QPU from the docs! and maybe also removing the converter.

Another option could be to simply have a `.. warning::` box that the converter is not support with the latest version of PL

**Review by josh146 — APPROVED** [review]
💯 Awesome, looks good from my end Antal!

File: doc/index.rst [hunk 3]
```diff
@@ -36,11 +36,6 @@ Currently, PennyLane-Forest provides these Forest devices for PennyLane:
     :description: Forest's QVM and pyQuil pyQVM simulator.
     :link: devices/qvm.html
 
-.. devicegalleryitem::
```
Comment (trbromley) [inline doc/index.rst:39]: We also mention QPUs once in the README:
> Contains the PennyLane Forest plugin. This plugin allows different Rigetti devices to work with PennyLane — the wavefunction simulator, the Quantum Virtual Machine (QVM), and Quantum Processing Units (QPUs).

Could just remove the last few words?

File: pennylane_forest/converter.py [hunk 4]
```diff
@@ -585,6 +587,10 @@ def load_program(program: pyquil.Program):
     Every variable that is present in the Program and that is not used as the target
     register of a measurement has to be provided in the ``parameter_map`` of the template.
 
+    .. warning::
+
+        Converting from Forest is not supported with PennyLane version 0.14.0.
```
Comment (trbromley) [inline pennylane_forest/converter.py:592]: Given that we raise an error above, is there any version of PL that this would work with?

File: pennylane_forest/converter.py (hunk 2 above)
Comment (trbromley) [reply pennylane_forest/converter.py:330]: Could we raise only if version 0.14 or above of PL?

File: pennylane_forest/qpu.py [hunk 5]
```diff
@@ -100,8 +100,9 @@ def __init__(
         calibrate_readout="plus-eig",
         **kwargs,
     ):
+        raise ValueError("Using the QPU via Forest is being deprecated.")
 
-        if readout_error is not None and load_qc:
+        if readout_error is not None and load_qc:  # pylint: disable=unreachable
```
Comment (trbromley) [inline pennylane_forest/qpu.py:105]: Why is this unreachable? :thinking:

File: tests/test_qpu.py [hunk 6]
```diff
@@ -66,7 +64,6 @@ def test_qpu_args(self):
         with pytest.raises(ValueError, match="Device has a fixed number of"):
             qml.device("forest.qpu", device=device, shots=5, wires=range(100), load_qc=False)
 
-    @flaky(max_runs=10, min_passes=3)
```
Comment (trbromley) [inline tests/test_qpu.py:69]: Do we need to remove flaky here? Is it to stop 10 repetitions of a test we know will fail?

**Review by trbromley — APPROVED** [review]
Thanks @antalszava, looks good to me! Just left some minor comments.

File: doc/index.rst (hunk 3 above)
Comment (antalszava) [reply doc/index.rst:39]: Good catch, thanks! Updated, also removed the device such that it's not provided by the plugin

File: pennylane_forest/converter.py (hunk 4 above)
Comment (antalszava) [reply pennylane_forest/converter.py]: Updated the message, it should work with PL version `<0.14.0`

File: pennylane_forest/qpu.py (hunk 5 above)
Comment (antalszava) [reply pennylane_forest/qpu.py:110]: There is an error that is raised before that

File: tests/test_qpu.py (hunk 6 above)
Comment (antalszava) [reply tests/test_qpu.py:69]: Yes, that was the original purpose, but rather placed them in a comment now.

File: pennylane_forest/converter.py (hunk 2 above)
Comment (antalszava) [reply pennylane_forest/converter.py]: Updated to do that 👍
